Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1534,6 +1553,33 @@ extension BrowserViewController: WKNavigationDelegate {
             // forward/backward. Strange, but LayoutChanged fixes that.
             UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil)
         }
+
+        addOpenInViewIfNeccessary(webView.URL)
+    }
+
+    private func addOpenInViewIfNeccessary(url: NSURL?) {
+        guard let url = url, let openInHelper = OpenInHelperFactory.helperForURL(url) else { return }
+        let view = openInHelper.openInView
+        webViewContainerToolbar.addSubview(view)
+        webViewContainerToolbar.snp_updateConstraints { make in
+            make.height.equalTo(OpenInViewUX.ViewHeight)
+        }
+        view.snp_makeConstraints { make in
+            make.edges.equalTo(webViewContainerToolbar)
+        }
+
+        self.openInHelper = openInHelper
+    }
+
+    private func removeOpenInView() {
+        guard let _ = self.openInHelper else { return }
+        let _ = webViewContainerToolbar.subviews.map { $0.removeFromSuperview() }
```

Comment:
Nit: you can use `forEach` here instead of `map` -- then you won't need the empty `let` assignment.